flink-tests/src/test/java/org/apache/flink/test/checkpointing/IgnoreInFlightDataITCase.java

```diff
@@ -255,7 +257,12 @@ public void cancel() {
 
         @Override
         public void snapshotState(FunctionSnapshotContext context) throws Exception {
-            if (lastCheckpointValue.get().get() > 0) {
+            if (context.getCheckpointId() > 2) {
+                // It is possible if checkpoint was triggered too fast after restart.
+                return; // Just ignore it.
+            }
+
+            if (context.getCheckpointId() == 2) {
                 throw new RuntimeException("Error during snapshot");
```

Comment:
nit: Could we adjust the message and maybe even use the `ExpectedTestException`? If I understand it correctly this is in most cases an expected failure, right?

Reply:
not a problem. done